Bulldozer.CSV/HelperMethods.cs

```diff
@@ -0,0 +1,502 @@
+﻿using Bulldozer.Model;
+using Bulldozer.Utility;
+using Rock;
+using Rock.Data;
+using Rock.Model;
+using Rock.Web.Cache;
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bulldozer.CSV
+{
+    partial class CSVComponent : BulldozerComponent
+    {
+
+        /// <summary>
+        /// Updates the person from person import and returns whether there were any changes to the person record
+        /// </summary>
+        /// <param name="lookupPerson">The lookup person.</param>
+        /// <param name="personImport">The person import.</param>
+        /// <param name="attributeValuesLookup">The attribute values lookup.</param>
+        /// <param name="familiesLookup">The families lookup.</param>
+        /// <param name="foreignSystemKey">The foreign system key.</param>
+        /// <param name="importDateTime">The import date time.</param>
+        /// <param name="recordTypeId">The RecordType Id to use.</param>
+        /// <returns></returns>
+        private bool UpdatePersonFromPersonImport( Person lookupPerson, PersonImport personImport, Dictionary<int, List<AttributeValueCache>> attributeValuesLookup, Dictionary<string, Group> familiesLookup, DateTime importDateTime, int recordTypeId )
+        {
+            using ( var rockContextForPersonUpdate = new RockContext() )
+            {
+                rockContextForPersonUpdate.People.Attach( lookupPerson );
+                var person = lookupPerson;
+
+                // Update Person
+                InitializePersonFromPersonImport( personImport, person, recordTypeId );
```

Comment:
This method returns errors. Should we be capturing them and doing anything with them?